src/dlgConnectionProfiles.cpp

```diff
@@ -1706,13 +1706,15 @@ void dlgConnectionProfiles::slot_connectToServer()
     }
 
     if (needsGenericPackagesInstall) {
-        //install generic mapper script
+        //install appropriate mapper script for the game
         if (pHost->getUrl().contains(QStringLiteral("aetolia.com"), Qt::CaseInsensitive) || pHost->getUrl().contains(QStringLiteral("achaea.com"), Qt::CaseInsensitive)
             || pHost->getUrl().contains(QStringLiteral("lusternia.com"), Qt::CaseInsensitive)
             || pHost->getUrl().contains(QStringLiteral("imperian.com"), Qt::CaseInsensitive)) {
             mudlet::self()->packagesToInstallList.append(QStringLiteral(":/mudlet-mapper.xml"));
         } else if (pHost->getUrl().contains(QStringLiteral("3scapes.org"), Qt::CaseInsensitive) || pHost->getUrl().contains(QStringLiteral("3k.org"), Qt::CaseInsensitive)) {
             mudlet::self()->packagesToInstallList.append(QStringLiteral(":/3k-mapper.xml"));
+        } else {
+            mudlet::self()->packagesToInstallList.append(QStringLiteral(":/mudlet-lua/lua/generic-mapper/generic_mapper.xml"));
```

Comment:
Is that the right path - or should the file be in the root of the source and resource file like the ***other two*** mapper files.   After all it is an XML file - not a Lua one...?

Reply:
I mean, I really think it should be moved and the path adjusted to be `":/generic_mapper.xml"`?

Oh, you now have a :beetle:! The file `./src/generic_mapper_script.xml` has been removed from the project but the record of it has NOT been removed from the `./src/mudlet.qrc` file so compilation is :100: failing!

Furthermore - there are **no** Lua files distributed in the resource file - in `:/mudlet-lua/lua/generic-mapper/` or elsewhere. Their management in \*nix packages is entirely handled by the CMake or QMake project files and the Makefile(s) both produce - they both have a list of files which they will copy to an install directory.  Indeed you can see this if you take a look at `mudlet.pro` in the region of this comment:
>\# Our customised 5.1 Lua Code Formatter files, unfortunately to get the Qt
>\# makefile constructor to reproduce the exact directory structure we have to
>\# list EVERY SINGLE B****Y FILE in a per directory grouping;

Also in `./src/CMakeLists.txt` right at the end is the \*nix installer bit - at the comment: `# *nix "make install" provision:`. Importantly, because CMake can do a wild-carded selection of file to include in an installation, I have included a safety step there to count the files selected for the Mudlet & Geyser and the LuaCodeFormatter parts of the installation in the parent `./CMakeLists.txt` - if the counts are wrong then a CMake warning message will be generated - so if you are fiddling with the files you might need to tweak the number of files!

Reply:
No, the path that it's right now at is good. If anything, we should move the other files into /lua as well.

I'll look into fixing the build.

Because this file is compiled into the binary, it does not need to be installed anywhere on the users system.

Reply:
But they are not ***lua*** files - it would be cleaner if all three (or indeed all the shipped `.xml` package files) were relocated to, say a `./src/xml` or `./src/packages` folder.

**\<spock_mode>** *That location currently being used - it is just not* logical, *Captain.***\</spock_mode>**

    
    
> Because this file is compiled into the binary, it does not need to be installed anywhere on the users system.

Maybe, but (unless an alias is used?) it has to be in that location in the source code.

Reply:
Okay - let's shuffle things around in a follow-up pr.